### PR TITLE
OGM-891 Use more natural format to store map-typed properties in Redis

### PR DESCRIPTION
**Pull request by mp911de**

### CODE REVIEW

File: core/src/main/java/org/hibernate/ogm/datastore/keyvalue/options/navigation/spi/BaseKeyValueStoreEntityContext.java [hunk 1]
```diff
@@ -22,4 +24,13 @@
 	public BaseKeyValueStoreEntityContext(ConfigurationContext context) {
 		super( context );
 	}
+
+	public E mapStorage(MapStorageType mapStorage) {
```
Comment (gunnarmorling) [inline core/src/main/java/org/hibernate/ogm/datastore/keyvalue/options/navigation/spi/BaseKeyValueStoreEntityContext.java]: I am a bit reluctant about this as a common option for all K/V stores. Can we expose it only for Redis at this point where we really support it? We still can pull it up later on.

File: documentation/manual/src/main/asciidoc/en-US/modules/redis.asciidoc [hunk 2]
```diff
@@ -1196,11 +1197,11 @@ Key: User:user_001
 { 
   "addresses" : [
     { 
-      "addresses_KEY" : "work",
+      "addressType" : "work",
```
Comment (gunnarmorling) [inline documentation/manual/src/main/asciidoc/en-US/modules/redis.asciidoc]: Is that actually right? The example seems not to re-define the map key column? Then by default indeed the `_KEY` suffix is appended.

File: redis/src/main/java/org/hibernate/ogm/datastore/redis/RedisDialect.java [hunk 3]
```diff
@@ -345,6 +385,42 @@ else if ( currentTtl != null && currentTtl > 0 ) {
 		return rows;
 	}
 
+	/**
+	 * Whether the rows of the given association should be stored in a hash using the single row key column as key or
+	 * not.
+	 */
+	private boolean organizeByRowKey(
```
Comment (gunnarmorling) [inline redis/src/main/java/org/hibernate/ogm/datastore/redis/RedisDialect.java]: I guess that could be extracted to a common place as well?

File: documentation/manual/src/main/asciidoc/en-US/modules/redis.asciidoc [hunk 4]
```diff
@@ -66,15 +66,6 @@ TTL can be configured on entity and association level (see <<ogm-redis-annotatio
 This property has no default value.
 hibernate.ogm.error_handler::
 The fully-qualified class name, class object or an instance of `ErrorHandler` to get notified upon errors during flushes (see <<ogm-api-error-handler>>)
-hibernate.ogm.datastore.redis.entity_storage::
-Defines the way OGM stores entity information in Redis.
-The following two strategies exist (values of the `org.hibernate.ogm.datastore.redis.options.EntityStorageType` enum):
-* `JSON` (store entities as JSON documents with a key) and
-* `HASH` (store entities within hashes. The fields and values are represented as strings within the hash).
```
Comment (gunnarmorling) [inline documentation/manual/src/main/asciidoc/en-US/modules/redis.asciidoc]: :)

**Comment by gunnarmorling** [comment]
@mp911de Added some minor remarks, but looks good to me overall.

File: core/src/main/java/org/hibernate/ogm/datastore/keyvalue/options/navigation/spi/BaseKeyValueStoreEntityContext.java (hunk 1 above)
Comment (mp911de) [inline core/src/main/java/org/hibernate/ogm/datastore/keyvalue/options/navigation/spi/BaseKeyValueStoreEntityContext.java]: Done.

File: documentation/manual/src/main/asciidoc/en-US/modules/redis.asciidoc (hunk 2 above)
Comment (mp911de) [inline documentation/manual/src/main/asciidoc/en-US/modules/redis.asciidoc]: Adjusted the example using the resulting JSON of `org.hibernate.ogm.datastore.redis.test.associations.MapMappingTest`

File: redis/src/main/java/org/hibernate/ogm/datastore/redis/RedisDialect.java (hunk 3 above)
Comment (mp911de) [inline redis/src/main/java/org/hibernate/ogm/datastore/redis/RedisDialect.java]: Moved it to `org.hibernate.ogm.datastore.document.impl.DotPatternMapHelpers`

**Comment by gunnarmorling** [comment]
Rebased and applied. Added two more commits for some minor fix ups. Thanks!

**Pull request closed without merging**